src/types.py

```diff
@@ -178,8 +185,14 @@ def intptr32() -> "Type":
         return Type(kind=Type.K_INTPTR, size=32, sign=Type.ANY_SIGN)
 
     @staticmethod
-    def ptr(type: Optional[Union["Type", CType]] = None) -> "Type":
-        return Type(kind=Type.K_PTR, size=32, sign=Type.ANY_SIGN, ptr_to=type)
+    def ptr(
+        type: Optional[Union["Type", CType]] = None, typemap: Optional[TypeMap] = None
+    ) -> "Type":
```

Comment:
I suspect this might nicely split into two methods `ptr(type: Optional[Type] = None)` and `cptr(type: CType, typemap: TypeMap)`, which gets rid of the need for the assert

Reply:
This is great, it helps tighten up the type bounds on the args to `Type.ptr`/`Type.cptr`